src/components/taglib/TransformHelper/assignComponentId.js

```diff
@@ -99,6 +99,12 @@ module.exports = function assignComponentId(isRepeated) {
                 el.setAttributeValue('id', this.buildComponentElIdFunctionCall(assignedKey));
             }
 
+            if (context.data.imperativeComponentIds &&  el.data.userAssignedKey !== false) {
+                if (context.data.imperativeComponentIds.includes(assignedKey.value)) {
```

Comment:
Just putting this here as a reminder than `Array.prototype.includes` is not supported in Node 4